gap/prop.gi

```diff
@@ -8,7 +8,47 @@
 #############################################################################
 ##
 
-#
+InstallMethod(IsMeetSemilatticeDigraph, "for a digraph",
+[IsDigraph],
+function(digraph)
+  local out_nbrs, i, j, intr;
+  if IsPartialOrderDigraph(digraph) then
+    out_nbrs := OutNeighbours(digraph);
+    for i in [1 .. Size(out_nbrs)] do
+      for j in [i + 1 .. Size(out_nbrs)] do
+        # Is there a better way of doing this? JDM
+        intr := Intersection(out_nbrs[i], out_nbrs[j]);
+        if not intr in out_nbrs then
```

Comment:
Also, seeing as every vertex has a loop, it is certainly true that if `intr = out_nbrs[i]` for some `i`, then that `i` must be one of the elements of `intr`. So with this method, instead of looking through the whole of `out_nbrs`, you can narrow down the search.